SweetTweetApp/src/main/webapp/resourses/js/App.js

```diff
@@ -170,30 +170,186 @@ const POSTS = [
 class App {
 
     constructor() {
-        this.tweetList = new TweetList(POSTS.map((p) => ({ ...p, createdAt: new Date(p.createdAt ) })));
+        //this.tweetList = new TweetList(POSTS.map((p) => ({...p, createdAt: new Date(p.createdAt)})));
+        this.tweetList = new TweetList();
+        this.tweetList.restore();
+        if (this.tweetList._posts.length === 0) {
+            this.tweetList.addAll(POSTS.map((p) => ({...p, createdAt: new Date(p.createdAt)})));
+        }
+        this.currentUser = localStorage.getItem("user");
+        this.view = new View();
+        this.viewPosts = null;
+        this.filters = null;
+        this.top = 10;
+        this.currentPage = "posts";
+        this.operation = null;
+        this.confirm = document.querySelector(".button-delete-post-on-field");
+        this.cancel = document.querySelector(".button-cancel");
+        this.message = document.querySelector(".field-for-deleting");
+        this.error = document.querySelector(".field-for-error");
+        this.loadMore = null;
+        this.closeButton = null;
+        this.filtering = null;
+        this.postForDeleting = null;
+        this.confirm.addEventListener("click", () => {
+            this.deletePost(this.postForDeleting);
+            this.message.classList.remove("emergence");
+        });
+        this.cancel.addEventListener("click", () => this.message.classList.remove("emergence"));
+    }
+
+    login(user) {
+        this.currentUser = user;
+        if (user) {
+            localStorage.setItem("user", user);
+        } else {
+            localStorage.removeItem("user");
+        }
+        this.setPage("posts");
```

Comment:
It can be nice to have smthng like enum with possible pages, or just move them to constants not to write it every time